### ISSUE TITLE
Dog not honoring force-fire command queue

### ISSUE THREAD

**Issue by SafwatHalaby** [OpenRA/OpenRA#14196]
Hold down CTRL (force-fire) + ALT (command queue/waypoint mode), then click several friendly infantry soldiers. 

The dog will only kill the first one, ignoring the rest of the queue.

This was not tested with enemies, but if the result is the same, it could hinder dog gameplay. (Suppose a player wants the dog to take out all riflemen first)

Release 20171014

**Comment by FRenzy-OpenRA** [comment]
It should work without force fire though ?

**Comment by abcdefg30** [comment]
> It should work without force fire though ?

Yes, it does work without.

**Comment by MunWolf** [comment]
This is how the trait seems to be implemented, it calls this
```
self.CancelActivity();
self.QueueActivity(new Leap(self, target.Actor, a, info.Speed, info.Angle));
```
which I presume cancels all queued activities.

**Comment by SafwatHalaby** [comment]
>It should work without force fire though ?

You cannot attack friendly units without force fire. I don't know whether or not the bug is present when attacking units with the command queue but without force fire. Testing dogs against the enemy AI is tricky.

**Comment by abcdefg30** [comment]
Closed by #15008 and #16007.
